crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -293,7 +348,17 @@ def upgrade_controller_nodes
         # and the existing cluster has been cleaned up by deleting most of resources
         return false unless node_api.post_upgrade
 
-        # FIXME: if upgrade went well, continue with next node(s)
+        # Delete router namespaces on non-upgraded node
+        master_node_api.delete_router_namespace
```

Comment:
@rhafer is is true this way we don't have much control over the router migration timing. What you propose sounds good, but I don't know how to face that implementation.